docs/DeveloperGuide.md

```diff
@@ -2,21 +2,198 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. Reference to AB-3 Developer Guide
+
+- [Source](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature)
+- Used as template to structure this Developer Guide
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+---
+
+### Ui component
+
+The `Ui` class handles message output for the user within the terminal.
+
+The `UI`component,
+- stores message output strings for user feedback
+- holds methods for printing messages to be called by other classes, such as `InputParsing`.
+
+#### Design considerations
+
+Being a CLI application, UI/UX is minimal and user IO is confined within the terminal.
+
+Hence, public methods within the`Ui` class are static for easy access by other classes, without the need to instantiate an instance of the `Ui` class to call.
+
+### InputParsing component
+
+This component ensures that the user parses in commands in a format that makes sense, which will modify the master list.
+
+![InputParsingUML](./diagrams/src/InputParsing/InputParsing.png)
+
+#### Design considerations
+- The parser must be effecive in breaking down the user's input into its constitutent commands, with further breakdown if an associated argument is added.
+- The parser should be quick and effective in understanding the user's input, with simple prompts given to help the user in correctly parsing the command through the input parser.
+- Allowing the user to input optional arguments. For example, the user could type `view <student name>`, which takes in the "student name" as an optional argument. This is to increase the robustness of the program, which accounts for the two types of users, one who just types in `view`, and the other as formerly mentioned above. 
+- Error handling must be intuitive for the user, so that appropriate error messages are produced if the user does not input a valid command. The error handling should also be robust, to account for the event a user is incapable in following instructions.
+
+#### Implementation and rationale
+
+![InputParsingSequenceDigram](./diagrams/src/InputParsing/InputParsingSequenceDiagram.png)
```

Comment:
Good try on the sequence diagram! However, should the studentInfo.txt be omitted from the sequence diagram? It would be better if there is a method call to the class which handles the modification of studentInfo.txt instead.